python/lsst/ip/isr/measureCrosstalk.py

```diff
@@ -316,17 +358,193 @@ def reduce(self, ratioList):
+        coeff, coeffErr, coeffNum = self.measureCrosstalkCoefficients(ratios, self.config.rejIter,
+                                                                      self.config.rejSigma)
         self.log.info("Coefficients:\n%s\n", coeff)
         self.log.info("Errors:\n%s\n", coeffErr)
         self.log.info("Numbers:\n%s\n", coeffNum)
         return coeff, coeffErr, coeffNum
 
-    def _getConfigName(self):
-        """Disable config output"""
-        return None
+    def measureCrosstalkCoefficients(self, ratios, rejIter=3, rejSigma=2.0):
+        """Measure crosstalk coefficients from the ratios.
 
-    def _getMetadataName(self):
-        """Disable metdata output"""
-        return None
+        Given a list of ratios for each target/source amp combination,
+        we measure a sigma clipped mean and error.
+
+        The coefficient errors returned are the standard deviation of
+        the final set of clipped input ratios.
+
+        Parameters
+        ----------
+        ratios : `list` of `list` of `numpy.ndarray`
+           Matrix of arrays of ratios.
+        rejIter : `int`
+           Number of rejection iterations.
+        rejSigma : `float`
+           Rejection threshold (sigma).
+
+        Returns
+        -------
+        coeff : `numpy.ndarray`
+            Crosstalk coefficients.
+        coeffErr : `numpy.ndarray`
+            Crosstalk coefficient errors.
+        coeffNum : `numpy.ndarray`
+            Number of pixels for each measurement.
+
+        Notes
+        -----
+        This has been moved into MeasureCrosstalkTask to allow for easier
+        debugging.
+
+        The lsstDebug.Info() method can be rewritten for __name__ =
+        `lsst.ip.isr.measureCrosstalk`, and supports the parameters:
+
+        debug.display['measure'] : `bool`
+            Display a histogram of the combined ratio measurements for
+            a pair of source/target amplifiers from the final set of
+            clipped input ratios.
+        """
+        if rejIter is None:
+            rejIter = self.config.rejIter
+        if rejSigma is None:
+            rejSigma = self.config.rejSigma
+
+        numAmps = len(ratios)
+        assert all(len(rr) == numAmps for rr in ratios)
+
+        coeff = np.zeros((numAmps, numAmps))
+        coeffErr = np.zeros((numAmps, numAmps))
+        coeffNum = np.zeros((numAmps, numAmps), dtype=int)
+
+        for ii, jj in itertools.product(range(numAmps), range(numAmps)):
+            if ii == jj:
+                values = [0.0]
+            else:
+                values = np.array(ratios[ii][jj])
+                values = values[np.abs(values) < 1.0]  # Discard unreasonable values
+
+            coeffNum[ii][jj] = len(values)
+
+            if len(values) == 0:
+                coeff[ii][jj] = np.nan
+                coeffErr[ii][jj] = np.nan
+            else:
+                if ii != jj:
+                    for rej in range(rejIter):
```

Comment:
This is an interesting approach. Is there a reason for not using afwMath stats and a clipped mean there? That would automatically do the number of iterations you want, and I believe would use the actual standard deviation, rather than the approximation from the IQR as is being done here.

If that's not desirable, could you just update the docstring to note that the sigma-clipping is using the IQR (times a const) rather than the std dev itself please?

Reply:
This is inherited code, but I believe the logic was that this is a noisy measurement, so using the Gaussian sigma approximation from the IQR gives an initial robust sigma estimate, allowing the clipping to be more aggressive.  I'll test the afwMath version to see if that's significantly different.

Reply:
I've decided against switching to `afwMath` simply because there is enough statistics code in `ip_isr` that this should probably be a full ticket itself (to identify what is in `afwMath`, and see what, if anything, would need to be added).

Reply:
OK, sounds good 👍